src/backend/expungeservice/expunger.py

```diff
@@ -131,6 +112,45 @@ def run(record: Record, today: date = date.today()) -> Dict[str, TimeEligibility
             ambiguous_charge_id_to_time_eligibility[charge.ambiguous_charge_id] = time_eligibility
         return ambiguous_charge_id_to_time_eligibility
 
+    @staticmethod
+    def _single_conviction_years_by_level(target_charge_level, relative_charge_date):
+        if "Felony Class A" in target_charge_level:
+            return (
+                date.max(),
+                "Never. Type ineligible charges are always time ineligible."
+            )
+        elif "Felony Class B" in target_charge_level:
+            return (
+                relative_charge_date + relativedelta(years=7),
+                "Seven years from date of conviction (137.225(1)(a))",
+            )
+        elif "Felony Class C" in target_charge_level:
+            return (
+                relative_charge_date + relativedelta(years=5),
+                "Five years from date of conviction (137.225(1)(a))",
+            )
+        elif "Misdemeanor Class A" in target_charge_level:
+            return (
+                relative_charge_date + relativedelta(years=3),
+                "Three years from date of conviction (137.225(1)(a))",
+            )
+        elif any([x in target_charge_level for x in ["Misdemeanor Class B", "Misdemeanor Class C", "Violation"]]):
+            return (
+                relative_charge_date + relativedelta(years=1),
+                "One year from date of conviction (137.225(1)(a))",
+            )
+        elif "Misdemeanor" in target_charge_level: # TODO: Is an unspecified Misdemeanor always a Class A?
```

Comment:
@michaelzhang43 Is an unspecified Misdemeanor always a Class A?

Reply:
Let's assume that it's Class A for practical purposes

Reply:
Not to be confused with the common situation in which a C Misdemeanor is listed as "C/Unclassified", in which case it should be considered a C Misdemeanor

Reply:
@michaelzhang43 that sounds surprising. Just to check, you mean the level would say "C/Unclassified" or would it say "Misdemeanor Class C/Unclassified"? (I mean, if it says "C/Unclassified" that just sounds awful if it's not specifying misdemeanor vs felony)